Source/displayinfo/include/ExtendedDisplayIdentification.h

```diff
@@ -124,22 +187,327 @@ namespace Plugin {
+                        for(uint8_t index = 0; index < dataBlock.BlockSize(); index++) {
+                            const uint8_t vic = dataBlock.Current()[index] & 0x7F;
+                            vicList.push_back(vic);
+                        }
+                    }
+                }
+            }
+
+            void SupportedAudioFormats(uint32_t& audioFormatMap)
+            {
+                DataBlockIterator dataBlock = DataBlockIterator(_segment, DetailedTimingDescriptorStart());
+                while(dataBlock.Next()) {
+                    if(dataBlock.IsValid() && (dataBlock.BlockTag() == 0x01)) {
+                        uint8_t sadCount = dataBlock.BlockSize() % 3;
+                        for(uint8_t index = 0; index < dataBlock.BlockSize(); index+=3) {
+                            const uint8_t sad = (dataBlock.Current()[index + 1] & 0x78) >> 3;
+                            // switch(((blockStart[sadStart + 1] & 0x78) >> 3)) {
+                            switch(sad){
+                            case 0x01:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::LPCM);
+                                break;
+                            case 0x02:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::AC3);
+                                break;
+                            case 0x03:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::MPEG1);
+                                break;
+                            case 0x04:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::MP3);
+                                break;
+                            case 0x05:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::MPEG2);
+                                break;
+                            case 0x06:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::AAC_LC);
+                                break;
+                            case 0x07:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::DTS);
+                                break;
+                            case 0x08:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::ATRAC);
+                                break;
+                            case 0x09:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::SUPER_AUDIO_CD);
+                                break;
+                            case 0x0A:
+                                audioFormatMap |= static_cast<uint32_t>(AudioFormatType::EAC3);
+                                // if MPEG surround implicitly and explicitly supported: assume ATMOS
+                                if((dataBlock.Current()[index + 3] & 0x01)) {
```

Comment:
Does this mean the EAC3 block is 6 bytes wide? If so shouldn't this be accounted for when iterating? Also you can be reaching beyond buffer her.